nest-forms-backend/src/nases/nases.service.ts

```diff
@@ -521,4 +532,52 @@ export default class NasesService {
       NasesErrorsResponseEnum.FILE_NOT_SCANNED,
     )
   }
+
+  public async sendToNasesAndUpdateState(
+    jwt: string,
+    form: Forms,
+    data: RabbitPayloadDto,
+    sender: SendMessageNasesSender,
+    additionalFormUpdates?: FormUpdateBodyDto,
+  ): Promise<void> {
+    const sendData = await this.nasesUtilsService.sendMessageNases(
+      jwt,
+      form,
+      sender,
+    )
+
+    if ((sendData && sendData.status !== 200) || !sendData) {
+      await this.formsService.updateForm(data.formId, {
+        state: FormState.DRAFT,
+        error: FormError.NASES_SEND_ERROR,
+      })
```

Comment:
If `this.nasesUtilsService.sendMessageNases()` throws, `this.formsService.updateForm()` won't happen. It would happen in the previous implementation, as it was in the catch of the upper function. It is implemented in such manner, that it won't throw, but it should either be explicitly mentioned here, or refactored further so that the error handling is uniformly try-catch